File: hto/54/hto_54.c
int res;
int *count;

int kthLargest(struct TreeNode* root, int k){
    count = &k;
    dfs(root, count);
    return res;
}

void dfs(struct TreeNode *node, int *k) {
    if (!node || (*k) == 0) return;
    dfs(node->right, k);
    (*k)--;
    if ((*k) == 0) {
        res = node->val;
        return;
    }
    dfs(node->left, k);
}

File: hto/54/hto_54.java
class Solution {
    int res, k;
    public int kthLargest(TreeNode root, int k) {
        this.k = k;
        dfs(root);
        return res;
    }

    private void dfs(TreeNode root) {
        if (root == null || k == 0) return;
        dfs(root.right);
        if (--k == 0) {
            res = root.val;
            return;
        }
        dfs(root.left);
    }
}